xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/descriptor/AbstractMacroDescriptor.java

```diff
@@ -181,16 +185,34 @@ public Map<String, ParameterDescriptor> getParameterDescriptorMap()
     @Override
     public String getDefaultCategory()
     {
-        return this.defaultCategory;
+        return this.defaultCategories.stream().findFirst().orElse(null);
```

Comment:
There's a problem here. It's a Set and thus has no order and you're returning the first item which could be different on different JVMs (unless you're using a special set). At the minimum it requires a comment to explain why it's going to work.

Reply:
Also, shouldn't this method be moved to legacy?

Reply:
It will, in a second step as it is used by `xwiki-platform` for now. I've added a todo here: https://github.com/xwiki/xwiki-rendering/pull/216/files#diff-09c9906c170a9fb45c33831886d7206217b9c49fa34d7c107c6335db0648e1b7R86